client/my-sites/upgrades/index.js

```diff
@@ -289,5 +290,14 @@ module.exports = function() {
 			controller.siteSelection,
 			upgradesController.checkout
 		);
+
+		page(
+			'/start-trial/:domain/:plan_name?',
```

Comment:
Then `plan_name` won't be optional but I'm find with this.
So `start-trial/:plan_name/:domain` ?
Whitelisted, you mean removing the `isEligible` middleware?